mona_openai/analysis/refusal.py

```diff
@@ -0,0 +1,112 @@
+"""
+Functionality for computing probability that a given GAI response is actually a refusal
+answer (which means the model can't or not allowed to answer the given prompt).
+"""
+from collections.abc import Iterable
+from os import environ
+
+from huggingface_hub.utils._errors import RepositoryNotFoundError
+from sentence_transformers import SentenceTransformer, util
+from torch import Tensor
+
+from ..exceptions import InvalidEmbeddingsModelName
+
+# Name of embedding model name to use.
+# In order to compute sentence embeddings, we use SentenceTransformers framework
+# (https://www.sbert.net/).
+# More information about the different models (and the tradeoff between speed and
+# performance of each one can be found here:
+# https://www.sbert.net/docs/pretrained_models.html)
+EMBEDDING_MODEL_NAME = environ.get("EMBEDDING_MODEL_NAME", "all-MiniLM-L6-v2")
+
+# Example sentences of LLM refusal answers (we will compare a given text to each one of
+# them in order to compute the probability that the given text is a refusal answer).
+REFUSAL_EXAMPLES = [
+    "I'm sorry, but I cannot assist with providing or discussing .",
+    "I'm unable to provide information or guidance on .",
+    "I cannot provide assistance or information on illegal activities, including but "
+    "not limited to ."
+    "I am prohibited from generating explicit, adult, or inappropriate content",
```

Comment:
maybe cut after "generating".

Reply:
Other examples as well, probably better to just keep the "refusal" part, and the not the specifics of the refusal, no?

Reply:
Once we have some tests in its own test file - it would be great to try to find things that break these and improve accordingly.